web/src/routes/(user)/share/[key]/+page.svelte

```diff
@@ -1,20 +1,74 @@
 <script lang="ts">
   import AlbumViewer from '$lib/components/album-page/album-viewer.svelte';
   import IndividualSharedViewer from '$lib/components/share-page/individual-shared-viewer.svelte';
-  import { SharedLinkType } from '@api';
+  import ControlAppBar from '$lib/components/shared-components/control-app-bar.svelte';
+  import ImmichLogo from '$lib/components/shared-components/immich-logo.svelte';
+  import ThemeButton from '$lib/components/shared-components/theme-button.svelte';
+  import Button from '$lib/components/elements/buttons/button.svelte';
+  import { api, SharedLinkType } from '@api';
   import type { PageData } from './$types';
+  import { handleError } from '$lib/utils/handle-error';
 
   export let data: PageData;
-  const { sharedLink } = data;
+  let { sharedLink, passwordRequired, sharedLinkKey: key } = data;
 
-  let isOwned = data.user ? data.user.id === sharedLink.userId : false;
+  let isOwned = !passwordRequired && data.user ? data.user.id === sharedLink?.userId : false;
+  let password = '';
+
+  const handlePasswordSubmit = async () => {
+    try {
+      const result = await api.sharedLinkApi.getMySharedLink({ password, key });
+      if (result.data) {
```

Comment:
I think this will always have data if the request doesn't throw an error. We don't check this anywhere else.